src/es6/SirenAction.js

```diff
@@ -151,6 +151,68 @@ const _performSirenAction = function(action, fields, tokenValue) {
 		});
 };
 
+const _combineActions = function(actionsAndFields) {
+	if (!actionsAndFields) return [];
+
+	// The structure of this is a map (hrefs) containing a map (methods) containing a map (fields)
+	const hrefsMap = new Map();
+
+	actionsAndFields.forEach(actionAndField => {
+		if (!actionAndField || !actionAndField.action) return;
+		const href = actionAndField.action.href;
+		const method = actionAndField.action.method;
+		if (!href || !method) return;
+
+		const methodsMap = hrefsMap.get(href) || new Map();
+		const fieldsMap = methodsMap.get(method) || new Map();
+
+		let fields;
+		if (actionAndField.fields) {
+			fields = _appendHiddenFields(actionAndField.action, actionAndField.fields);
+		} else {
+			fields = _getSirenFields(actionAndField.action);
+		}
+		fields.forEach(field => fieldsMap.set(field.name, field.value));
+
+		methodsMap.set(method, fieldsMap);
+		hrefsMap.set(href, methodsMap);
+	});
+
+	const combinedActions = [];
+	for (const [href, methodsMap] of hrefsMap.entries()) {
```

Comment:
This could probably be built in place to avoid the second loop, but OK to leave for now.
If we get some unit tests in a follow up PR :-) then it should be easy to refactor this kind of thing.

Reply:
I guess for the fields at least though, using the map in the first loop makes it easier to manage (ignore?/overwrite?) duplicate field names. As you are translating the mapped fields into objects in the second loop.

Reply:
Currently we are losing any other field information in this process, e.g. `type` but the only place we currently seem to use that is in the `_performSirenAction._addHiddenFields` which I think will be called again on the aggregated action, only there will never be any hidden fields which is kind of by definition at that point. But something for us to be aware of for future changes maybe.